packages/adapter-knex/lib/adapter-knex.js

```diff
@@ -869,10 +872,13 @@ class QueryBuilder {
             );
           }
 
+          const subSubQuery = listAdapter.parentAdapter.knex.raw(
+            `SELECT "${selectCol}" FROM (${subQuery}) AS unused_alias`
```

Comment:
Why do you need an alias?

Reply:
https://www.postgresql.org/docs/12/sql-select.html

> A sub-SELECT can appear in the FROM clause. This acts as though its output were created as a temporary table for the duration of this single SELECT command. Note that the sub-SELECT must be surrounded by parentheses, and an alias must be provided for it. A VALUES command can also be used here.